src/main/java/de/vogel612/testclient_javabot/TestProgram.java

```diff
@@ -39,14 +39,23 @@
 
 	public TestProgram() {
 		LOGGER.info("instantiating TestProgram class");
-		AppContext.INSTANCE.add(chatInterface);
-		bot = new DefaultBot(chatInterface);
-		gui = new ClientGui();
-		chatInterface.subscribe(bot);
-		chatInterface.subscribe(gui);
-		LOGGER.info("Basic component setup completed, beginning command glueing.");
-		new CoreBotCommands(chatInterface, bot).allCommands().forEach(bot::subscribe);
-		LOGGER.info("Glued Core Commands");
+        try{
+	        AppContext.INSTANCE.add(chatInterface);
+	        bot = new DefaultBot(chatInterface);
+	        Executors.newSingleThreadExecutor().execute(() -> {
```

Comment:
It shuts down when the main window is closed. JavaFX calls the stop() (http://docs.oracle.com/javase/8/javafx/api/javafx/application/Application.html#stop--) once you close the main window. Though I have seen other threads still running after the main window is closed, which can be looked into. These threads doesn't include the thread which is indicated by @Zomis here.